prusti-server/src/backend.rs

```diff
@@ -0,0 +1,27 @@
+use prusti_common::{
+    vir::{LoweringContext, ToViper},
+    Stopwatch,
+};
+use viper::{VerificationContext, VerificationResult};
+
+pub enum Backend<'a> {
+    Viper(viper::Verifier<'a>, &'a VerificationContext<'a>),
+}
+
+impl<'a> Backend<'a> {
+    pub fn verify(&mut self, program: &prusti_common::vir::program::Program) -> VerificationResult {
+        match self {
+            Backend::Viper(viper, context) => {
+                let mut stopwatch = Stopwatch::start("prusti-server", "verifier startup");
+
+                let ast_factory = context.new_ast_factory();
+                let viper_program = program.to_viper(LoweringContext::default(), &ast_factory);
+
```

Comment:
When constructiong a Viper program, we also need to conditionally dump it for debugging:
```rust
if config::dump_viper_program() {
    stopwatch.start_next("dumping viper program");
    dump_viper_program(
        &ast_utils,
        viper_program,
        program.get_name_with_check_mode(),
    );
}
```